src/Components/ComparePublicationData/ComparePublicationDataModal.tsx

```diff
@@ -224,31 +223,93 @@ const ComparePublicationDataModal: FC<ComparePublicationDataModalProps> = ({
             }
       );
     };
-    //TODO! NB! this should be removed as soon as contributors does not use localstorage anymore
-    window.localStorage.removeItem('tempContributors');
     initFormik().then();
-    getContributors().then();
   }, [isDuplicate, state.selectedPublication, importPublication]);
 
-  async function fetchAllAuthors(resultId: string) {
-    if (state.doSave) {
-      //TODO: dosave-check her også ?
-      let page = 1;
-      let allAuthors: ContributorType[] = [];
-      while (allAuthors.length < state.selectedPublication.authorTotalCount) {
-        const contributorResponse = await getContributorsByPublicationCristinResultId(
-          resultId,
-          page,
-          500,
-          SearchLanguage.Nb // Gir det mening med språk her ?
-        );
-        allAuthors = [...allAuthors, ...contributorResponse.data];
-        page++;
+  useLayoutEffect(() => {
+    async function enrichImportPublicationAuthors() {
+      const tempContributors: ContributorWrapper[] = [];
+      try {
+        setIsLoadingContributors(true);
+        setLoadingContributorsError(undefined);
+        const identified: boolean[] = [];
+        const authorsFromImportPublication = importPublication.authors;
+
+        let cristinAuthors: ContributorType[];
+        if (isDuplicate) {
+          cristinAuthors = state.selectedPublication.authors;
+        } else {
+          cristinAuthors = await searchCristinPersons(authorsFromImportPublication);
+        }
+        for (let i = 0; i < Math.max(cristinAuthors.length, authorsFromImportPublication.length); i++) {
+          if (cristinAuthors[i]) {
+            if (isDuplicate && state.doSave) {
+              if (i < cristinAuthors.length) {
+                cristinAuthors[i].affiliations = await getDuplicateAffiliations(state.selectedPublication.authors[i]);
+              } else {
+                cristinAuthors[i] = { ...emptyContributor };
+              }
+            }
+            identified[i] = cristinAuthors[i].identified_cristin_person || false;
+            tempContributors[i] = createContributorWrapper(authorsFromImportPublication, i, cristinAuthors);
+            tempContributors[i].toBeCreated = await generateToBeCreatedContributor(
+              tempContributors[i],
+              cristinAuthors[i],
+              authorsFromImportPublication[i],
+              isDuplicate
+            );
+          }
+        }
+        dispatch({ type: 'setContributorsLoaded', payload: true });
+        dispatch({ type: 'identified', payload: identified }); //skjer dette to steder ?
+        dispatch({ type: 'identifiedImported', payload: identified });
+        validateContributors(tempContributors, setContributorErrors);
+      } catch (error) {
+        setLoadingContributorsError(error as Error);
+      } finally {
+        setIsLoadingContributors(false);
```

Comment:
Skulle denne funksjonen vært splittet opp i for løkka slik at vi lager promise.array? Da går hentinga raskere.

Reply:
Høres lurt ut. lager en egen task for det.